### PR TITLE
Updates CI to fix broken checks and builds caused by WASM introduction

### PR DESCRIPTION
**Pull request by ibeckermayer**
- Updates the `.github/workflows/lint-ui.yaml` and `.github/workflows/unit-tests-ui.yaml` GHA to use the Teleport buildbox container. This allows these CI runs to include a `Build WASM` step using [`wasm-pack`](https://github.com/rustwasm/wasm-pack), which is required for building the rust-based WASM module. This WASM module is required for the linter and UI tests to pass.
- Add `babel-plugin-transform-import-meta` and `babel-plugin-transform-vite-meta-env` to the babel config. This is necessary for `import.meta.url` and `import.meta.env` (respectively) to work in a nodejs environment (which is to say, for the `jest` tests to work).
- Changes all `from 'teleport/ironrdp/pkg'` to `from 'teleport/ironrdp/pkg/ironrdp'` It's not entirely clear to me why this is necessary, it's likely some discrepancy between how vite does things vs how nodejs wants things. This is an easy enough fix and it works (no need to get fancy with further babel magic).
- Updates README with some information on WASM.

Related PRs:
- https://github.com/gravitational/teleport.e/pull/2944
- https://github.com/gravitational/teleport/pull/28105
- https://github.com/gravitational/teleport/pull/35731

### CODE REVIEW

File: package.json
Comment (camscale) [inline package.json:48]: your editor seems to be removing newlines at the end of files. Can you fix that do it doesn't do that? Every line should end with a newline, including the last (see https://unix.stackexchange.com/questions/18743/whats-the-point-in-adding-a-new-line-to-the-end-of-a-file)

File: web/packages/teleport/package.json
Comment (camscale) [inline web/packages/teleport/package.json:35]: same here

File: package.json
Comment (camscale) [inline package.json:16]: Adding a new build tool requirement is something that should be called out loudly. It seems rather hidden in this PR titled "Fix yarn test" where the PR description does not mention it either. I'm not sure of the relationship between the rust `wasm-pack` package/module and the yarn `build-wasm` command so I think that should be documented somewhere somehow too.

File: web/packages/teleport/package.json
Comment (camscale) [inline web/packages/teleport/package.json:32]: Not sure if these deps were added manually or automatically, but if manually I find it is best to keep lists like this in alphabetical order. It makes it easy to know when you have duplicates (or more accurately you don't get duplicates because it is obvious) and it helps manage merge conflicts when things get added at different places rather than all at the end. If these were added automatically by some command, please ignore this comment.

Comment (ibeckermayer) [reply web/packages/teleport/package.json]: a80054fdfaaec1b3ad8eb44e632baca3b2173a2f

File: web/packages/teleport/package.json
Comment (ibeckermayer) [reply web/packages/teleport/package.json]: a80054fdfaaec1b3ad8eb44e632baca3b2173a2f

File: package.json
Comment (ibeckermayer) [reply package.json]: These were removed in favor of running them as a separate step in the CI pipeline https://github.com/gravitational/teleport/pull/35401/commits/9a3579c00fd89c4253d93d545ea06d1bc75820cd

This is because we don't necessarily want to ensure wasm is built before _every_ run of `yarn test`, e.g. sometimes we just want to do `yarn test <specific test which may not even rely on wasm modules>

**Review by probakowski — APPROVED** [review]

File: Makefile
Comment (camscale) [inline Makefile:14]: Ensure this change does not get merged please.

What I usually do is create a `camh/test/foo` branch for `camh/foo` changes and run the test build from there - it keeps the PR branch clean and you just switch back to it and delete the test branch to clean up. The tags still hang around though. We should consider cleaning up dev tags every now and then.

**Review by camscale — COMMENTED** [review]
It's a bit hard to review this with all the changes from `make update-version`. I've suggested a way to avoid that, but please ensure you don't merge this with the results of running that.

File: .eslintignore
Comment (ibeckermayer) [reply .eslintignore]: It needn't be in the gitignore because wasm-pack always generates this directory with a gitignore inside with

```
*
```

I did remove the ** from here and .prettierignore and everything still works.

File: Makefile
Comment (ibeckermayer) [reply Makefile]: Good call, I'll do that in the future. Should be polished for review now.

**Comment by ibeckermayer** [comment]
In theory every build pipeline is passing now. In practice I had one fail on me due to a GHA error:

image

https://drone.platform.teleport.sh/gravitational/teleport/31857/10/1
https://github.com/gravitational/teleport.e/actions/runs/7225580694

However I have another run where that succeeded and I haven't made relevant changes since:
image
https://drone.platform.teleport.sh/gravitational/teleport/31839/6/2
https://github.com/gravitational/teleport.e/actions/runs/7216993875

Other runs are failing for reasons unrelated to what I'm doing.

Therefore once the related `e` changes have merged (https://github.com/gravitational/teleport.e/pull/2944) and this PR's e-ref updated to the merged commit, this should be good to go.

File: Makefile
Comment (zmb3) [inline Makefile:14]: Don't forget to revert this.

File: e
Comment (zmb3) [inline e:1]: Don't forget to update this after the `e` PR merges.

File: web/packages/teleterm/src/agentCleanupDaemon/agentCleanupDaemon.test.ts
Comment (zmb3) [inline web/packages/teleterm/src/agentCleanupDaemon/agentCleanupDaemon.test.ts:89]: Where is said `--init` flag?

**Review by zmb3 — APPROVED** [review]

File: e
Comment (camscale) [inline e:1]: the commit you are updating `e` to point to is not on master on teleport.e - this should not be merged to teleport master

File: web/README.md
Comment (camscale) [inline web/README.md:122]: s/modul/module/

File: build.assets/Makefile
Comment (camscale) [inline build.assets/Makefile:85]: The comments about release-unix-preserving-webassets etc should probably be attached to this line. It is far from where it is used so should have some comment at least.

File: web/README.md
Comment (camscale) [inline web/README.md:122]: This PR does not include anything in `web/packages/teleport/src/ironrdp` and there is nothing there on master. I'm a little confused how this PR works without this source code. Does this README change belong in a different PR?

File: web/packages/teleterm/src/agentCleanupDaemon/agentCleanupDaemon.test.ts
Comment (ibeckermayer) [reply web/packages/teleterm/src/agentCleanupDaemon/agentCleanupDaemon.test.ts:89]: It's used here:

https://github.com/gravitational/teleport/pull/35401/files#diff-6c099f0fc9d8f8a6677998cdb8f0f619e662a887bf91adbe15d7a2dc5af57ebbR22

Comment (zmb3) [reply web/packages/teleterm/src/agentCleanupDaemon/agentCleanupDaemon.test.ts:89]: Gotcha. Any reason why the large comment explaining why the flag is necessary is so far away from where the flag is used? Would it make more sense to move this reasoning over there?

File: Makefile
Comment (ibeckermayer) [reply Makefile]: e63a6e2c59025fae385617c86d82742e6d8fc525

Comment (ibeckermayer) [reply Makefile]: e63a6e2c59025fae385617c86d82742e6d8fc525

File: build.assets/build-webassets-if-changed.sh
Comment (ibeckermayer) [reply build.assets/build-webassets-if-changed.sh]: 🧙‍♂️

3103c30f844a4b0faab5dfd7a54e250e01347b64

File: build.assets/Makefile
Comment (ibeckermayer) [reply build.assets/Makefile:89]: See 1de16ff8d8b47eb3abcbe89c57729ce328d56fa1

File: web/README.md
Comment (ibeckermayer) [reply web/README.md]: 72f0aac4b8637a1263c53e72d203fc598f25bb74

Comment (ibeckermayer) [reply web/README.md]: This PR is being merged into the `ironrdp` branch, which is the big feature branch for the IronRDP client refactor:

https://github.com/gravitational/teleport/pull/26874

**Review by camscale — APPROVED** [review]
LGTM. I'm not sure how this gets into master in the end and how long that takes. It's quite possible we'll make changes to the same makefile targets as we clean up, so these changes will conflict when the ironrdp branch is being merged to master.

File: web/packages/teleterm/src/agentCleanupDaemon/agentCleanupDaemon.test.ts
Comment (ibeckermayer) [reply web/packages/teleterm/src/agentCleanupDaemon/agentCleanupDaemon.test.ts:89]: I understood the need for that flag in close relation to the behavior of this specific test, so wrote it here in order to codify that. IMO I would need to link to this test in order to properly document why `init` was added over in `unit-tests-ui.yaml`, so I figured I may as well leave the explanation where it is and [just link to explanation + test](https://github.com/gravitational/teleport/pull/35401/files#diff-6c099f0fc9d8f8a6677998cdb8f0f619e662a887bf91adbe15d7a2dc5af57ebbR20-R21).

**Comment by ibeckermayer** [comment]
> LGTM. I'm not sure how this gets into master in the end and how long that takes. It's quite possible we'll make changes to the same makefile targets as we clean up, so these changes will conflict when the ironrdp branch is being merged to master.

May god have mercy on our souls

**Pull request merged**
